File: switch5g/app/src/main/java/com/dongp/switch5g/BootReceiver.java
package com.dongp.switch5g;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * 恢复内置定时的时机：
 * - 开机完成
 * - 小米的快速重启（QUICKBOOT_POWERON）
 * - 本应用被重新安装/更新（MY_PACKAGE_REPLACED，此时闹钟也会失效）
 */
public class BootReceiver extends BroadcastReceiver {

    private static final String ACTION_QUICKBOOT = "android.intent.action.QUICKBOOT_POWERON";

    @Override
    public void onReceive(Context c, Intent intent) {
        String action = intent == null ? null : intent.getAction();
        if (action == null
                || (!Intent.ACTION_BOOT_COMPLETED.equals(action)
                && !ACTION_QUICKBOOT.equals(action)
                && !Intent.ACTION_MY_PACKAGE_REPLACED.equals(action))) {
            return;
        }
        Scheduler.scheduleNext(c, BuildConfig.MODE);
    }
}

File: haorizi/app/src/main/java/com/awu/haorizi/BootReceiver.kt
package com.awu.haorizi

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

/**
 * 开机 / 改系统时间 / 版本更新后，把闹钟重新排一遍。
 * 数据从 SharedPreferences 读，不依赖 WebView —— 开机时不会有 JS 环境。
 */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action ?: return
        Log.i(ReminderScheduler.TAG, "BootReceiver: $action，恢复提醒")
        try {
            ReminderScheduler.reschedule(context)
        } catch (e: Exception) {
            Log.e(ReminderScheduler.TAG, "恢复提醒失败: ${e.message}")
        }
    }
}
